File: src/test/java/ownDataRepresentation/PartitionedNodeListFileTest.java
package ownDataRepresentation;

import de.esymetric.jerusalem.ownDataRepresentation.Node;
import de.esymetric.jerusalem.ownDataRepresentation.fileSystem.LatLonDir;
import de.esymetric.jerusalem.ownDataRepresentation.fileSystem.PartitionedNodeListFile;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class PartitionedNodeListFileTest {

    @Test
    public void testPartionedNodeListFile() throws Exception {
        String dataDirectoryPath = "testData";
        PartitionedNodeListFile nlf = new PartitionedNodeListFile(dataDirectoryPath, false);

        int nodeID = nlf.insertNewNodeStreamAppend(48.11, 11.48);

        int nodeID2 = nlf.insertNewNodeStreamAppend(48.111, 11.481);

        int nodeID3 = nlf.insertNewNodeStreamAppend(-48.111, -11.481);


        nlf.close();

        // ---

        nlf = new PartitionedNodeListFile(dataDirectoryPath, false);

        nlf.setTransitionID(48.111, 11.481, nodeID2, 2222);
        nlf.setTransitionID(48.11, 11.48, nodeID, 1111);
        nlf.setTransitionID(-48.111, -11.481, nodeID3, 3333);

        nlf.close();

        // ---

        nlf = new PartitionedNodeListFile(dataDirectoryPath, true);
        Node n1 = nlf.getNode(new LatLonDir(48.11, 11.48), nodeID);
        assertEquals(1111, n1.transitionID);

        Node n2 = nlf.getNode(new LatLonDir(48.111, 11.481), nodeID2);
        assertEquals(2222, n2.transitionID);

        Node n3 = nlf.getNode(new LatLonDir(-48.111, -11.481), nodeID3);
        assertEquals(3333, n3.transitionID);

        nlf.close();
    }
}

File: src/test/java/ownDataRepresentation/PartitionedNodeListFileTest.kt
package ownDataRepresentation

import de.esymetric.jerusalem.ownDataRepresentation.fileSystem.LatLonDir
import de.esymetric.jerusalem.ownDataRepresentation.fileSystem.PartitionedNodeListFile
import org.junit.Assert
import org.junit.Test

class PartitionedNodeListFileTest {
    @Test
    @Throws(Exception::class)
    fun testPartionedNodeListFile() {
        val dataDirectoryPath = "testData"
        var nlf = PartitionedNodeListFile(dataDirectoryPath, false)
        val nodeID = nlf.insertNewNodeStreamAppend(48.11, 11.48)
        val nodeID2 = nlf.insertNewNodeStreamAppend(48.111, 11.481)
        val nodeID3 = nlf.insertNewNodeStreamAppend(-48.111, -11.481)
        nlf.close()

        // ---
        nlf = PartitionedNodeListFile(dataDirectoryPath, false)
        nlf.setTransitionID(48.111, 11.481, nodeID2, 2222)
        nlf.setTransitionID(48.11, 11.48, nodeID, 1111)
        nlf.setTransitionID(-48.111, -11.481, nodeID3, 3333)
        nlf.close()

        // ---
        nlf = PartitionedNodeListFile(dataDirectoryPath, true)
        val n1 = nlf.getNode(LatLonDir(48.11, 11.48), nodeID)
        Assert.assertEquals(1111, n1!!.transitionID.toLong())
        val n2 = nlf.getNode(LatLonDir(48.111, 11.481), nodeID2)
        Assert.assertEquals(2222, n2!!.transitionID.toLong())
        val n3 = nlf.getNode(LatLonDir(-48.111, -11.481), nodeID3)
        Assert.assertEquals(3333, n3!!.transitionID.toLong())
        nlf.close()
    }
}